osidb/mixins.py

```diff
@@ -79,3 +81,49 @@ def clean(self):
 
     class Meta:
         abstract = True
+
+
+class AlertMixin(models.Model):
+    """
+    This mixin implements the necessary mechanisms to have validation alerts.
+
+    The way that this mixin works is simple, any model that inherits from this mixin
+    will have a field in which alerts are stored in JSON, this field is re-populated
+    on each save when the validations are run.
+
+    The mixin provides a helper function for creating said alerts, this also serves
+    as an abstraction layer to enforce a schema on the JSON field and guarantee that
+    the alerts are somewhat constant in their content.
+    """
+
+    _alerts = models.JSONField(default=dict)
```

Comment:
What about adding some alert getter? The underscored attribute would be an unusual interface.

Reply:
Don't think we need to, there should ideally be only one case in which the attribute is accessed for read: exposing over API, and in that case it would be wrapped in a getter at the API level (something like `ReadOnlyField`)